dataland-e2etests/src/test/kotlin/org/dataland/e2etests/tests/frameworks/Sme.kt

```diff
@@ -1,47 +1,75 @@
+    val customSmeDataControllerApi = CustomSmeDataControllerApi(keycloakToken)
+    val testDataProviderForSmeData = FrameworkTestDataProvider(SmeData::class.java)
+
+    val testSmeData = testDataProviderForSmeData.getTData(1).first()
+
+    val dummyFileAlpha = File("dummyFileAlpha.txt")
+    val dummyFileBeta = File("dummyFileBeta.txt")
+
+    @BeforeAll
+    fun prepareDummyFiles() {
+        val threeMegabytes = 3 * 1000 * 1000
+        val tenMegabytes = 10 * 1000 * 1000
+
+        dummyFileAlpha.writeBytes(ByteArray(threeMegabytes))
+        dummyFileBeta.writeBytes(ByteArray(tenMegabytes))
+    }
+
+    @AfterAll
+    fun deleteDummyFiles() {
+        dummyFileAlpha.delete()
+        dummyFileBeta.delete()
     }
 
     @Test
-    fun `post a company with SME data and check if the data can be retrieved correctly`() {
-        val listOfUploadInfo = apiAccessor.uploadCompanyAndFrameworkDataForOneFramework(
-            listOfOneCompanyInformation,
-            listOfOneSmeDataSet,
-            apiAccessor::smeUploaderFunction,
-        )
-        val receivedDataMetaInformation = listOfUploadInfo[0].actualStoredDataMetaInfo
-        val downloadedAssociatedData = apiAccessor.dataControllerApiForSmeData
-            .getCompanyAssociatedSmeData(receivedDataMetaInformation!!.dataId)
-        val downloadedAssociatedDataType = apiAccessor.metaDataControllerApi
-            .getDataMetaInfo(receivedDataMetaInformation.dataId).dataType
-
-        Assertions.assertEquals(receivedDataMetaInformation.companyId, downloadedAssociatedData.companyId)
-        Assertions.assertEquals(receivedDataMetaInformation.dataType, downloadedAssociatedDataType)
-
-        Assertions.assertEquals(
-            sortNaturalHazardsCovered(listOfOneSmeDataSet[0]),
-            sortNaturalHazardsCovered(downloadedAssociatedData.data),
+    fun `post a company with SME data including documents and check if it has been persisted successfully`() {
+        val companyId = apiAccessor.uploadOneCompanyWithRandomIdentifier().actualStoredCompany.companyId
+        val companyAssociatedDataSmeData = CompanyAssociatedDataSmeData(companyId, "2022", testSmeData)
+
+        val initialDataMetaInfo = customSmeDataControllerApi.postCompanyAssociatedDataSmeData(
+            companyAssociatedDataSmeData,
+            listOf(dummyFileAlpha, dummyFileBeta),
         )
+
+        Thread.sleep(2000) // Wait required to give the asynchronous EuroDaT storage process enough time to finish
+
+        val persistedDataMetaInfo =
+            apiAccessor.metaDataControllerApi.getDataMetaInfo(initialDataMetaInfo.dataId)
+
+        assertEquals(QaStatus.Pending, initialDataMetaInfo.qaStatus)
+        assertEquals(false, initialDataMetaInfo.currentlyActive)
+
+        assertEquals(QaStatus.Accepted, persistedDataMetaInfo.qaStatus)
+        assertEquals(true, persistedDataMetaInfo.currentlyActive)
+
+        assertEquals(persistedDataMetaInfo.dataId, initialDataMetaInfo.dataId)
```

Comment:
Kann man hier nicht die Objekte direkt vergleichen?

Reply:
Nein weil der QualityStatus und der Aktivitätsstatus sich nach Persistierung ändern und daher ungleich sind in beiden